subprojects/core/src/test/groovy/org/gradle/api/internal/AbstractDomainObjectCollectionSpec.groovy

```diff
@@ -34,6 +34,8 @@ abstract class AbstractDomainObjectCollectionSpec<T> extends Specification {
 
     abstract T getD()
 
+    abstract boolean isInsertionOrderExpected()
```

Comment:
the idea of the `iterationOrder()` was to encapsulate this, so given the elements in the order inserted, it would return them in the expected iteration order. This way we don't need a flag or conditional stuff in each of the test cases.

Reply:
Yeah, this was just because we didn't have iteration order retained on `ListElementSource`.  I went ahead and implemented that, too, and removed these conditionals.